test/JDBC/input/BABEL-3640.sql

```diff
@@ -0,0 +1,22 @@
+use master
+go
+
+create table trans2(id int identity(1,1) primary key, source int, target int, amount int);
+go
+
+declare @p0 sysname = null;
+declare @p1 sysname = null;
+declare @p2 sysname = null;
+declare @p3 varchar(32) = '''''''TABLE''''''';
+EXEC sp_tables @P0, @P1, @P2, @P3; -- non-empty
+go
+
+declare @p0 sysname = null;
+declare @p1 sysname = null;
+declare @p2 sysname = null;
+declare @p3 varchar(32) = '''TABLE''';
+EXEC sp_tables @P0, @P1, @P2, @P3; -- empty
```

Comment:
1. Could you check T-SQL behavior for this as well?
2. Please add a test for `'TABLE'` case.

Reply:
I confirmed this behavoir is the same with SQL server